### PR TITLE
fix: add storage_limit check

### LINKED ISSUE #18091: Validators for Harbor APIs
In order to improve user experience and strengthen security, let's make the validators of the front and back ends consistent:
- [ ] 1:  Post `/api/2.0/projects`,    `storage_limit`  should  range  from -1 to 1024TiB
- [ ] 2: For all cron string,    the first field should always be `0`
- [x] 3: Cron string for replication rule, the second field can not be `*`
- [x] 4: Tag name, see #18073
- [x] 5: Secret length for robot account, should range from 8 to 128
- [x] 6: P2p-preheat policy, `name` should follow the pattern `^[a-z0-9]+(?:[._-][a-z0-9]+)*$`

### PR DESCRIPTION
**Pull request by zyyw**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #18091
task item: 1

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Review by AllForNothing — APPROVED** [review]
lgtm

File: src/pkg/quota/types/resources.go [hunk 1]
```diff
@@ -25,6 +25,9 @@ const (
 	// UNLIMITED unlimited resource value
 	UNLIMITED = -1
 
+	// MaxLimitedValue the max capacity of limited storage, in bytes
+	MaxLimitedValue = uint64(1125899906842624)
```
Comment (wy65701436) [inline src/pkg/quota/types/resources.go:29]: why is it the MaxLimitedValue of quota?

File: src/pkg/config/manager.go [hunk 2]
```diff
@@ -188,7 +190,27 @@ func (c *CfgManager) ValidateCfg(ctx context.Context, cfgs map[string]interface{
 		if item.Scope == metadata.SystemScope {
 			return fmt.Errorf("system config items cannot be updated, item: %v", key)
 		}
+
 		strVal := utils.GetStrValueOfAnyType(value)
+
+		// check storage per project before setting it
+		if key == common.StoragePerProject {
```
Comment (wy65701436) [inline src/pkg/config/manager.go:197]: why put the specific checking at here?

File: src/pkg/quota/types/resources.go (hunk 1 above)
Comment (zyyw) [reply src/pkg/quota/types/resources.go:29]: 1125899906842624 Bytes is 1024 TB.

File: src/pkg/config/manager.go (hunk 2 above)
Comment (zyyw) [reply src/pkg/config/manager.go:197]: Because the check is only for `common.StoragePerProject`

**Review by stonezdj — APPROVED** [review]

File: src/pkg/config/manager.go [hunk 3]
```diff
@@ -188,7 +190,27 @@ func (c *CfgManager) ValidateCfg(ctx context.Context, cfgs map[string]interface{
...
+			storagePerProject, err := strconv.ParseInt(strVal, 10, 64)
+			if err != nil {
+				return fmt.Errorf("cannot parse string value(%v) to int64", strVal)
+			}
+			if storagePerProject <= 0 {
```
Comment (stonezdj) [inline src/pkg/config/manager.go:202]: Because same code appear twice, sugest to extract to function

Comment (zyyw) [reply src/pkg/config/manager.go]: thanks for the review. It has been refactored.

**Pull request merged**